### DISCUSSION TITLE
Export Object separately

### DISCUSSION THREAD

**Discussion by umanda** [fabricjs/fabric.js#7131] [Q&A]
I can export entire canvas data into Raster image, SVG or JSON. However, I have a requirement to export each object in canvas separately, as layers. For example, If I have 2 SVG objects placed in Fabricjs canvas I would like to export them separately into a Raster image or SVG file.  Is there by native way to do that? or if someone can suggest any idea would be really appreciated.

**Comment by akalana** [comment]
I am also suffering from this issue. Anyone knows then please support to resolve this issue.

**Reply by umanda** [reply]
Hello Nipuna

If you found any clue please kindly share here.

**Comment by asturur** [comment]
So there are a couple of ways, depending how you want to export them.
Can you be more specific? do you need something that is big as your canvas, and contains only one of the object, in the position where it is in the canvas, or the export has to contain ONLY the object and be as small as the object?
Would be nice if you can make some practical example.

**Reply by umanda** [reply]
Hello Andrea
Thank you very much for helping me.

Let's imagine I have a canvas 800x800 and I have placed five objects as following attachment. I would like to export them separately keeping the x,y coordination along with the canvas.

![image](https://user-images.githubusercontent.com/368338/122783084-deeb7180-d2ce-11eb-90d8-af8c94a44b73.png)

After the process, I will end up with five items ( SVG, png or jpg) which are having exact same width and height of canvas and relative to the position.

**Comment by i-prikot** [comment]
Hi.
I think you can turn on, turn off the parameter `visible` one by one, or move one element at a time to a `staticCanvas` and save.
http://fabricjs.com/docs/fabric.Object.html#visible

**Reply by umanda** [reply]
Hello i-prikot. 

I will try, thank you very much. If I will success will make a demo and will share it here. Thank you again.

**Reply by asturur** [reply]
Yes visible is an easy way to go for this.
